src/components/OptionsList/BaseOptionsList.tsx

```diff
@@ -1,89 +1,72 @@
 import React, {forwardRef, memo, useEffect, useRef} from 'react';
+import type {SectionListData, SectionListRenderItem} from 'react-native';
 import {View} from 'react-native';
-import _ from 'underscore';
 import OptionRow from '@components/OptionRow';
 import OptionsListSkeletonView from '@components/OptionsListSkeletonView';
 import SectionList from '@components/SectionList';
 import Text from '@components/Text';
 import usePrevious from '@hooks/usePrevious';
 import useThemeStyles from '@hooks/useThemeStyles';
+import type {OptionData} from '@libs/ReportUtils';
 import variables from '@styles/variables';
 import CONST from '@src/CONST';
-import {defaultProps as optionsListDefaultProps, propTypes as optionsListPropTypes} from './optionsListPropTypes';
+import type {BaseOptionListProps, OptionsList, Section} from './types';
 
-const propTypes = {
-    /** Determines whether the keyboard gets dismissed in response to a drag */
-    keyboardDismissMode: PropTypes.string,
-
-    /** Called when the user begins to drag the scroll view. Only used for the native component */
-    onScrollBeginDrag: PropTypes.func,
-
-    /** Callback executed on scroll. Only used for web/desktop component */
-    onScroll: PropTypes.func,
-
-    /** List styles for SectionList */
-    listStyles: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.object), PropTypes.object]),
-
-    ...optionsListPropTypes,
-};
-
-const defaultProps = {
-    keyboardDismissMode: 'none',
-    onScrollBeginDrag: () => {},
-    onScroll: () => {},
-    listStyles: [],
-    ...optionsListDefaultProps,
-};
-
-function BaseOptionsList({
-    keyboardDismissMode,
-    onScrollBeginDrag,
-    onScroll,
-    listStyles,
-    focusedIndex,
-    selectedOptions,
-    headerMessage,
-    isLoading,
-    sections,
-    onLayout,
-    hideSectionHeaders,
-    shouldHaveOptionSeparator,
-    showTitleTooltip,
-    optionHoveredStyle,
-    contentContainerStyles,
-    sectionHeaderStyle,
-    showScrollIndicator,
-    listContainerStyles: listContainerStylesProp,
-    shouldDisableRowInnerPadding,
-    shouldPreventDefaultFocusOnSelectRow,
-    disableFocusOptions,
-    canSelectMultipleOptions,
-    shouldShowMultipleOptionSelectorAsButton,
-    multipleOptionSelectorButtonText,
-    onAddToSelection,
-    highlightSelectedOptions,
-    onSelectRow,
-    boldStyle,
-    isDisabled,
-    innerRef,
-    isRowMultilineSupported,
-    isLoadingNewOptions,
-    nestedScrollEnabled,
-    bounces,
-    renderFooterContent,
-}) {
+function BaseOptionsList(
+    {
+        keyboardDismissMode = 'none',
+        onScrollBeginDrag = () => {},
+        onScroll = () => {},
+        listStyles = [],
+        focusedIndex = 0,
+        selectedOptions = [],
+        headerMessage = '',
+        isLoading = false,
+        sections = [],
+        onLayout,
+        hideSectionHeaders = false,
+        shouldHaveOptionSeparator = false,
+        showTitleTooltip = false,
+        optionHoveredStyle = undefined,
```

Comment:
No need for undefined | null default values, we should be able to handle that with ?. and ?? operators